packages/timeline-state-resolver/src/devices/vizMSE.ts

```diff
@@ -982,13 +982,15 @@ class VizMSEManager extends EventEmitter {
 						await this._initializeShows(uniqueShowIds)
 
 						setTimeout(() => {
-							elementHashesToDelete.map(async (elementHash) => {
-								const element = this._elementCache[elementHash]
-								if (element?.toDelete) {
-									await this._deleteElement(element.content)
-									delete this._elementCache[elementHash]
-								}
-							})
+							Promise.all(
+								elementHashesToDelete.map(async (elementHash) => {
+									const element = this._elementCache[elementHash]
+									if (element?.toDelete) {
+										await this._deleteElement(element.content)
+										delete this._elementCache[elementHash]
+									}
+								})
+							).catch((error) => this.emit('error', error))
```

Comment:
I think this is okay. This way you should get the stack trace, which won't be there if you'll `toString` an `Error`. And pepTalk errors, which this will most likely be, are well described as well.